TCAT/Model/Route.swift

```diff
@@ -159,7 +159,7 @@ class Route: NSObject, Codable {
         for (index, direction) in rawDirections.enumerated() {
             if index != last && index != first && direction.type == .walk {
                 direction.type = .arrive
-                direction.name = rawDirections[index - 1].endLocation.name
+                direction.name = rawDirections[index - 1].stops.last?.name ?? "Nil"
```

Comment:
nitpick: While this should never appear, I think "Bus Stop" would look be a bit better of an error state than "Nil" to show users